crates/starknet-server/tests/test_advancing_time.rs

```diff
@@ -72,22 +67,177 @@ mod advancing_time_tests {
+            calldata: vec![],
+        };
+        devnet
+            .json_rpc_client
+            .call(call_current_timestamp.clone(), BlockId::Tag(BlockTag::Latest))
+            .await
+            .unwrap()[0]
+    }
+
+    #[tokio::test]
+    async fn timestamp_syscall_set_in_past() {
+        let devnet: BackgroundDevnet =
+            BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let new_contract_address = setup_timestamp_contract(&devnet).await;
+
+        // set time in past
+        let past_time = 1;
+        let set_time_body = Body::from(json!({ "time": past_time }).to_string());
+        let resp_set_time = devnet.post_json("/set_time".into(), set_time_body).await.unwrap();
+        let resp_body_set_time = get_json_body(resp_set_time).await;
+        assert_eq!(resp_body_set_time["block_timestamp"], past_time);
+
+        // mine block
+        devnet.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
+
+        // check if timestamp is greater/equal
+        let current_timestamp = get_current_timestamp(&devnet, new_contract_address).await;
+        assert_ge_with_buffer(current_timestamp.to_string().parse::<u64>().ok(), Some(past_time));
+    }
+
+    #[tokio::test]
+    async fn timestamp_syscall_set_in_future() {
+        let now = get_unix_timestamp_as_seconds();
+        let devnet: BackgroundDevnet =
+            BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let new_contract_address = setup_timestamp_contract(&devnet).await;
+
+        // set time in future
+        let future_time = now + 1000;
+        let set_time_body = Body::from(json!({ "time": future_time }).to_string());
+        let resp_set_time = devnet.post_json("/set_time".into(), set_time_body).await.unwrap();
+        let resp_body_set_time = get_json_body(resp_set_time).await;
+        assert_eq!(resp_body_set_time["block_timestamp"], future_time);
+
+        // mine block
+        devnet.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
+
+        // check if timestamp is greater/equal
+        let current_timestamp = get_current_timestamp(&devnet, new_contract_address).await;
+        assert_ge_with_buffer(current_timestamp.to_string().parse::<u64>().ok(), Some(future_time));
+    }
+
+    #[tokio::test]
+    async fn timestamp_syscall_increase_time() {
+        let now = get_unix_timestamp_as_seconds();
+        let devnet: BackgroundDevnet =
+            BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let new_contract_address = setup_timestamp_contract(&devnet).await;
+
+        // increase time
+        let increase_time: u64 = 1000;
+        let increase_time_body = Body::from(json!({ "time": increase_time }).to_string());
+        devnet.post_json("/increase_time".into(), increase_time_body).await.unwrap();
+
+        // check if timestamp is greater/equal
+        let mut current_timestamp = get_current_timestamp(&devnet, new_contract_address).await;
```

Comment:
Instead of having a mutable `current_timestamp`, why not have to different variables, which can then also be asserted as in the next test (`timestamp_syscall_contract_constructor`) - that the second one is greater.

Reply:
fixed in 5b8dba1ad980cd3310fd4a14ed78eb636e12b8ea